lib/client.js

```diff
@@ -251,6 +251,12 @@ Resolve.prototype._fillServers = function() {
 
   slist = platform.name_servers;
 
+  if (slist.length === 0) {
+    // No nameservers defined on platform, so this._server_list
+    // stays empty.
+    return;
+  }
```

Comment:
This doesn't seem like the right way to fix this...? (I could be mistaken.)

Shouldn't the fix be applied to the while loop condition instead?

Why is it using `platform.attempts` in the loop condition, for example?

cc @bobvanderlinden @tjfontaine @silverwind

(Sorry folks, this is all my volunteer time! Could use your help with this!)